### PR TITLE
Update contributing guidelines

### PR DESCRIPTION
**Pull request by tiziano88**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,17 +22,42 @@ use GitHub Pull Requests (PR) for this purpose. Consult
 [GitHub Help](https://help.github.com/articles/about-pull-requests/) for more
 information on using PRs.
 
+For a new review, the author works on a new branch on their own fork of the
+repository (we don't create additional branches on the main repository).
+
+Once the changes are made locally, the author creates a new commit on their
+newly created branch based on master, then pushes to their own fork, and creates
+a PR from the GitHub UI. The author then selects one or more reviewers.
```
Comment (daviddrysdale) [inline CONTRIBUTING.md]: Might be a bit tricky for an external contributor to select a reviewer?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,17 +22,42 @@ use GitHub Pull Requests (PR) for this purpose. Consult
...
+
+The commit / PR description should follow
+[standard git commit conventions](https://chris.beams.io/posts/git-commit/).
```
Comment (daviddrysdale) [inline CONTRIBUTING.md]: nit: s/git/Git/

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -22,17 +22,42 @@ use GitHub Pull Requests (PR) for this purpose. Consult
...
+PR is merged via the
 ["Squash and merge"](https://help.github.com/en/articles/about-pull-request-merges#squash-and-merge-your-pull-request-commits)
-button in the GitHub UI. The final commit message is taken from the PR
-description, and individual commit messages are discarded; for this reason, we
-do not expect commits to have meaningful messages during a review; the PR
-description should follow
-[standard git commit conventions](https://chris.beams.io/posts/git-commit/).
+button in the GitHub UI. The UI will suggest a final commit message composed of
+the PR title, and individual commit messages as a bullet point list; the author
+can then reword the final commit message in the UI, usually discarding the
```
Comment (daviddrysdale) [inline CONTRIBUTING.md]: s/can/should/

File: CONTRIBUTING.md (hunk 1 above)
Comment (tiziano88) [reply CONTRIBUTING.md]: Yeah, hopefully some of us will notice though.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tiziano88) [reply CONTRIBUTING.md]: Done.

File: CONTRIBUTING.md (hunk 3 above)
Comment (tiziano88) [reply CONTRIBUTING.md]: Done.

**Pull request merged**